crates/shared/src/sources/balancer_v2/swap.rs

```diff
@@ -229,11 +249,36 @@ impl BaselineSolvable for StablePoolRef<'_> {
         )
         .ok()?;
         let amount_in_before_fee = in_reserves.downscale_up(in_amount).ok()?;
-        let in_amount = add_swap_fee_amount(amount_in_before_fee, self.swap_fee).ok()?;
+        add_swap_fee_amount(amount_in_before_fee, self.swap_fee).ok()
+    }
 
-        converge_in_amount(in_amount, out_amount, |x| {
```

Comment:
Is there a particular reason this was moved out of that function? It looks like it could have stayed there which wouldn't have the risk of forgetting to call this function in `get_amount_in()`.

Reply:
I moved this out as I wanted the `regular_swap_*` functions to be the "pure" swap logic while the guarantees from converging in amounts is particular to the `BaselineSolvable` trait and not the pool itself.